### ISSUE TITLE
Offload accessibility and positioning burden of select components to 3rd party libraries

### ISSUE THREAD

**Issue by vincerubinetti** [monarch-initiative/monarch-app#96]
Right now, I'm using vue-tippy for tooltip position, accessibility, and styling. But for multi select components, which also have dropdowns/popovers, I'm doing all of that manually. With positioning, I'm doing it very simply, with no sliding/flipping/etc when the dropdown goes beyond the screen.

Two main libraries to look at:

- the brand new and cleanly refactored [floating ui](https://floating-ui.com/docs/react-dom-interactions). used for correctly positioning tooltips/popovers/menus/etc. might also be usable for accessibility in the future, if they ever make a Vue version of their [`use` React hooks](https://floating-ui.com/docs/useListNavigation). might also be used to replace my custom modal code as well.
- [vue combo blocks](https://github.com/sssmi/vue-combo-blocks), a Vue version of the React library [downshift](https://www.downshift-js.com/use-combobox) which has a very convenient and clean way to get all the accessibility props you need for a select component. it doesn't have many users at the moment, but there must be an alternative. and as stated above, hopefully floating ui will implement this soon, so we only need one library for all of this.

**Comment by vincerubinetti** [comment]
This is definitely necessary I think.

I'm finding that the [implementation examples](https://www.w3.org/TR/2021/NOTE-wai-aria-practices-1.2-20211129/examples) I've been modeling off of, from the W3 themselves, are not in line with the spec, and are also throwing errors with Axe (which seems to follow the spec very closely, i.e. I trust its warnings/errors more than the W3 examples). 

I'm interested to see if downshift and vue-combo-blocks will pass Axe tests no problem. From what I can see, they'll do a better job than the examples.
